src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -612,14 +623,21 @@ discover_daos_mount(void)
 		return;
 	}
 
-	/* the length of fs_root has been checked already. */
-	strncpy(dfs_list[num_dfs].fs_root, fs_root, DFS_MAX_PATH);
+	/* The length of fs_root has been checked already.
+	 * Use strndup() here. D_STRNDUP() leads to NLT errors.
+	 * Not sure whether it is due to D_FREE() called in destructor function or not.
+	 */
```

Comment:
Without being able to check Jenkins NLT errors will be about a suspected memory leak I think.  Each allocation must have a matching free, so D_STRUNDUP should then be freed with D_FREE.

Reply:
Yes. I did have paired of D_STRUNDUP() and D_FREE(). D_FREE() is called inside destructor function. Could this cause NLT failure?

Reply:
Do you have a link to the NLT issue?  The only one I can see in Jenkins is https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12218/11/testReport/junit/NLT/test/Unit_Tests___NLT_on_EL_8___test_pil4dfs/ and that was the result of a segfault.

We have the macros for a reason and the entire rest of the DAOS tree manages to use them, please do so and if this results in test failures then either look into the failures or seek help.

Reply:
Thank you! You are right. I found the issue and it should be fixed with the new commit.